### PR TITLE
created non-root user for finemapping docker

### PR DESCRIPTION
**Pull request by RobinM-code**
this fixes the docker-snakemake permission issue for rule: run finemapping in otg-data-loading...

**note that after this pull request is accepted, the finemapping config file: finemapping.yaml in otg-data-loading should be changed!**
finmapping folder in input locations and output locations paths should be changed from:
`/finemapping/`
to
`/home/my-user/finemapping/`

### CODE REVIEW

File: Dockerfile [hunk 1]
```diff
@@ -39,4 +51,11 @@ RUN curl https://sdk.cloud.google.com | bash
...
-COPY ./ /finemapping
\ No newline at end of file
+COPY ./ /home/my-user/finemapping
+
+# Make all files in finemapping owned by the non-root user
+USER root
+RUN chown -R my-user:my-user /home/my-user/finemapping
+
+# Run container as non-root user
+USER my-use
```
Comment (romanhaa) [inline Dockerfile]: I suppose this should be `my-user`, right?

**Review by romanhaa — COMMENTED** [review]
Can we use a different name instead of`my-user`, e.g. `otg`? I feel like that would be a little more intuitive.

I'll try this setup on my machine.

File: Dockerfile (hunk 1 above)
Comment (RobinM-code) [reply Dockerfile]: yep

**Comment by romanhaa** [comment]
It works on my machine with the expected adjustments (paths in `finemapping.yaml` and in the `run_finemapping` rule). I think this PR is good to go.

**Review by romanhaa — APPROVED** [review]

**Comment by romanhaa** [comment]
We discovered that the new Docker image generated (significantly) different results compared to the previous. For the two test studies, I noticed the number of variants in the credible set go down from ~5,400 to ~200. I had a brief look at the Conda/Python package versions in the old and new images and saw that there at least a few differences. The idea that package version can be responsible for such a large difference in results is quite scary. I think we should further investigate before adopting the new Docker image.

**Comment by romanhaa** [comment]
I made two test runs. The new Docker image generates the same numbers of credible sets as the original (old) image. So I think we are all set now. Still have no idea what caused the issue and why it's fixed now.

**Pull request merged**